src/libxc/CMakeLists.txt

```diff
@@ -0,0 +1,153 @@
+	gga_xc_edf1.c gga_xc_1w.c
+	gga_c_optc.c gga_c_tca.c
+	gga_x_bpccac.c gga_x_sogga11.c
+	gga_c_sogga11.c gga_xc_th1.c
+	gga_x_wpbeh.c gga_x_hjs.c
+	gga_x_ityh.c gga_x_sfat.c
+	gga_x_ev93.c gga_x_ak13.c
+	gga_x_q2d.c gga_c_q2d.c
+	gga_x_ssb_sw.c gga_c_ft97.c
+	gga_x_n12.c gga_x_lv_rpw86.c
+	gga_x_2d_b86.c gga_x_2d_b86_mgc.c
+	gga_x_2d_b88.c gga_x_2d_pbe.c
+	gga_k_tflw.c gga_k_pearson.c
+	gga_k_ol1.c gga_k_ol2.c
+	gga_k_dk.c gga_k_meyer.c
+	gga_x_vmt.c gga_x_hcth_a.c
+	gga_x_cap.c gga_x_beefvdw.c
+	gga_x_sg4.c gga_x_gg99.c
+	gga_x_pbepow.c gga_x_lag.c
+	gga_k_thakkar.c gga_xc_oblyp_d.c
+	hyb_gga_xc_b3lyp.c hyb_gga_xc_o3lyp.c
+	gga_x_pbeint.c gga_x_rge2.c
+	gga_x_vmt84.c gga_x_hjs_b88_v2.c
+	gga_c_revtca.c gga_c_bcgp.c
+	gga_c_zpbeint.c gga_c_pbeloc.c
+	gga_c_sg4.c gga_c_regtpss.c
+	gga_c_scan_e0.c gga_xc_th2.c
+	gga_xc_th3.c gga_c_hcth_a.c
+	hyb_gga_xc_wb97.c gga_c_bmk.c
+	gga_c_gapc.c gga_c_gaploc.c
+	gga_c_zvpbeint.c gga_c_w94.c
+	gga_c_cs1.c gga_c_op_xalpha.c
+	gga_c_op_b88.c gga_c_op_pbe.c
+	gga_c_op_g96.c gga_c_op_pw91.c
+	gga_x_pbetrans.c gga_x_chachiyo.c
+	gga_k_exp4.c hyb_gga_xc_pbeh.c
+	hyb_gga_xc_b1wc.c hyb_gga_xc_hse.c
+	hyb_gga_xc_cam_b3lyp.c
+	hyb_gga_xc_camy_blyp.c
+	hyb_gga_xc_camy_b3lyp.c hyb_gga_xc_lcy_pbe.c
+	hyb_gga_xc_lcy_blyp.c gga_xc_vv10.c
+	hyb_gga_xc_edf2.c mgga.c
+	mgga_x_lta.c mgga_x_tpss.c
+	mgga_x_br89.c mgga_x_gvt4.c
+	mgga_x_m06l.c mgga_x_tau_hcth.c
+	mgga_x_2d_prhg07.c mgga_x_pkzb.c
+	hyb_mgga_x_m05.c mgga_x_m08.c
+	mgga_x_ms.c mgga_x_mn12.c
+	mgga_x_mk00.c mgga_c_vsxc.c
+	mgga_c_pkzb.c mgga_c_bc95.c
+	mgga_c_m08.c mgga_xc_cc06.c
+	mgga_c_cs.c mgga_xc_otpss_d.c
+	mgga_xc_zlp.c hyb_mgga_xc_b88b95.c
+	mgga_x_mbeef.c mgga_xc_b97mv.c
+	mgga_x_scan.c mgga_c_scan.c
+	hyb_mgga_xc_tpssh.c mgga_x_mvs.c
+	hyb_mgga_x_mvsh.c mgga_x_tm.c
+	mgga_x_vt84.c mgga_x_sa_tpss.c
+	mgga_k_pc07.c mgga_x_mbeefvdw.c
+	mgga_x_m11.c mgga_x_m11_l.c
+	hyb_mgga_xc_wb97mv.c mgga_c_m05.c
+	mgga_c_m06l.c hyb_mgga_x_dldf.c
+	mgga_c_tpss.c mgga_c_revtpss.c
+	mgga_c_tpssloc.c mgga_xc_hle17.c
+	mgga_c_kcis.c mgga_xc_lp90.c
+	hyb_mgga_xc_kcis.c mgga_c_b88.c
+	mgga_x_gx.c mgga_x_pbe_gx.c
+	mgga_c_revscan.c mgga_x_br89_explicit.c
+	mgga_xc_b98.c xc_f.c)
+
+set(LIBXC_FORTRAN_SOURCES
+	libxc_funcs.f90
+	libxc.f90)
+
+set(LIBXC_GPU_SOURCES
+	gpu.c)
+
+# -------------------------------------------------------------
+# enable optimization
+set_property(SOURCE ${LIBXC_C_SOURCES} PROPERTY COMPILE_FLAGS ${OPT_CFLAGS_SPC})
+set_property(SOURCE ${LIBXC_FORTRAN_SOURCES} PROPERTY COMPILE_FLAGS ${OPT_FFLAGS_SPC})
+
+# -------------------------------------------------------------
+# Generate fortran mangling header
+include(FortranCInterface)
+FortranCInterface_HEADER(${CMAKE_CURRENT_BINARY_DIR}/xc-fortran-mangling.h)
+
+# -------------------------------------------------------------
+# create libraries
+
+add_library(xc STATIC ${LIBXC_C_SOURCES} ${LIBXC_FORTRAN_SOURCES})
+make_pic_if_needed(xc)
+target_include_directories(xc PUBLIC . ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Isn't ${CMAKE_CURRENT_SOURCE_DIR} automatically included? By the way, why to include ${CMAKE_CURRENT_BINARY_DIR} as well?

Reply:
No, no directories are automatically included by default.